pkg/amp/encryption_key.go

```diff
@@ -0,0 +1,210 @@
+	log.Debug("Initializing Encryption Key, %q", cfg.Name())
+	k := &encryptionKey{
+		EncryptionKey: cfg,
+		keyManagement: km,
+	}
+
+	var err error
+	k.providerEncryptionKey, err = prov.NewEncryptionKey(k, cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return k, nil
+}
+
+// Route satisfies the embedded route.Router interface in resource.EncryptionKey.
+// EncryptionKey handles load, create, destroy, audit, config and info requests by delegating them
+// to the providerEncryptionKey.
+func (k *encryptionKey) Route(req *route.Request) route.Response {
+	log.Route(req, "Encryption Key %q", k.Name())
+	switch req.Command() {
+	case route.Create:
+		if err := k.Create(req.Flags().Get()...); err != nil {
+			msg.Error(err.Error())
+			return route.FAIL
+		}
+		return route.OK
+	case route.Destroy:
+		if err := k.Destroy(req.Flags().Get()...); err != nil {
+			msg.Error(err.Error())
+			return route.FAIL
+		}
+		return route.OK
+	case route.Provision:
+		if err := k.Provision(req.Flags().Get()...); err != nil {
+			msg.Error(err.Error())
+			return route.FAIL
+		}
+		return route.OK
+	case route.Audit:
+		if err := k.Audit("Encryption Key"); err != nil {
+			msg.Error(err.Error())
+			return route.FAIL
+		}
+		return route.OK
+	case route.Info:
+		k.Info()
+		return route.OK
+	case route.Config:
+		k.Print()
+		return route.OK
+	default:
+		panic("Internal Error: Unknown command " + req.Command().String())
+	}
+}
+
+func (k *encryptionKey) Load() error {
+	return k.providerEncryptionKey.Load()
+}
+
+func (k *encryptionKey) Create(flags ...string) error {
+	if k.Created() {
+		msg.Detail("Encryption Key exists, skipping...")
+		return nil
+	}
+	if err := k.providerEncryptionKey.Create(flags...); err != nil {
+		return err
+	}
+	if err := k.createSecurityTags(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Created satisfies the embedded resource.Creator interface in resource.Bucket.
+// It delegates the call to the provider's encryptionKey.
+func (k *encryptionKey) Created() bool {
+	return k.providerEncryptionKey.Created()
+}
+
+func (k *encryptionKey) Destroy(flags ...string) error {
+	if k.Destroyed() {
+		msg.Detail("Encryption Key does not exist, skipping...")
+		return nil
+	}
+	return k.ProviderEncryptionKey().Destroy(flags...)
+}
+
+// Destroyed satisfies the embedded resource.Destroyer interaface in resource.Bucket.
+// It delegates the call to the provider's encryptionKey.
+func (k *encryptionKey) Destroyed() bool {
+	return k.providerEncryptionKey.Destroyed()
+}
+
+func (k *encryptionKey) Provision(flags ...string) error {
+	if k.Destroyed() {
+		msg.Detail("Encryption Key does not exist, skipping...")
+		return nil
+	}
+	if err := k.createSecurityTags(); err != nil {
+		return err
+	}
+	return k.ProviderEncryptionKey().Provision(flags...)
+}
+func (k *encryptionKey) Audit(flags ...string) error {
+	if len(flags) == 0 || flags[0] == "" {
```

Comment:
Do you need to create an audit session if it doesn't already exist. I am assuming that you can do: amp <acct> key foo audit, and get to this code without a session already existing.